apps/server/src/state.ts

```diff
@@ -0,0 +1,456 @@
+import { EndAction, OntimeEvent, Playback, TimerLifeCycle, TimerState, TimerType } from 'ontime-types';
+import { clock } from './services/Clock.js';
+import { RestorePoint, restoreService } from './services/RestoreService.js';
+import { calculateDuration, dayInMs } from 'ontime-utils';
+import { getCurrent, getExpectedFinish } from './services/timerUtils.js';
+import type { DeepReadonly, DeepWritable } from 'ts-essentials';
+import { eventStore } from './stores/EventStore.js';
+import { integrationService } from './services/integration-service/IntegrationService.js';
+import { PlaybackService } from './services/PlaybackService.js';
+import { updateRoll } from './services/rollUtils.js';
+
+type TState = DeepReadonly<{
+  playback: Playback;
+  timer: TimerState & {
+    pausedTime: number;
+    pausedAt: number | null;
+    timeEnd: number | null;
+    finishedNow: boolean;
+    lastUpdate: number | null;
+    secondaryTarget: number | null;
+  };
+}>;
+
+export const state: TState = {
+  playback: Playback.Stop,
+  timer: {
+    clock: clock.timeNow(),
+    current: null,
+    elapsed: null,
+    expectedFinish: null,
+    addedTime: 0,
+    pausedTime: 0,
+    pausedAt: null,
+    timeEnd: null,
+    startedAt: null,
+    finishedAt: null,
+    secondaryTimer: null,
+    selectedEventId: null,
+    duration: null,
+    timerType: null,
+    endAction: null,
+    lastUpdate: null,
+    secondaryTarget: null,
+    get finishedNow() {
+      return this.current <= 0 && this.finishedAt === null;
+    },
+  },
+};
+
+export const stateMutations = {
+  timer: {
```

Comment:
I wonder if removing this timer object will be better in the long run since the idea is that all state is treated as a holistic piece of data.

In the cases we have, playback mutations dont make sense to be separed from timer mutation, the same is true for the next piece of state to move (loaded)